testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2545,6 +2545,14 @@ public void testDeleteWithSubquery()
             assertUpdate("DELETE FROM " + table.getName() + " WHERE EXISTS(SELECT 1 WHERE false)", 0);
             assertUpdate("DELETE FROM " + table.getName() + " WHERE EXISTS(SELECT 1)", "SELECT count(*) - 1 FROM orders");
         }
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete_correlated_exists_subquery", "AS SELECT * FROM nation")) {
+            // delete using correlated EXISTS subquery
+            assertUpdate(format("DELETE FROM %1$s WHERE EXISTS(SELECT regionkey FROM region WHERE regionkey = %1$s.regionkey AND name LIKE 'A%%')", table.getName()), 15);
+            assertQuery(
+                    "SELECT * FROM " + table.getName(),
+                    "SELECT * FROM nation WHERE regionkey IN (SELECT regionkey FROM region WHERE name NOT LIKE 'A%')");
+        }
```

Comment:
Are we testing engine or a connector here?
i.e. does this belong to BCT?

Reply:
Bump.

I think we're testing engine only here. Please convince me otherwise, or move it out of BCT.

Reply:
But for specific patterns of a query - should it be in BCT or ?

Reply:
BCT should exercise connectors and their interactions with the engine.

for a thing that's not really related to a particular connector, `AbstractTestEngineOnlyQueries` can be used.

I think we may need a variant of `AbstractTestEngineOnlyQueries` with a connector that supports deletes and updates (Iceberg preferred).

Reply:
I agree that it should move to `AbstractTestEngineOnlyQueries` but it is kind of a broader change as we need to migrate most of the similar tests related to Delete with a sub-queries to `AbstractTestEngineOnlyQueries` from `BCT`. 

> I think we may need a variant of AbstractTestEngineOnlyQueries with a connector that supports deletes and updates (Iceberg preferred).

Yeah. This can be supported as a part of the migration (which could be a different PR). WDYT ? As of now `AbstractTestEngineOnlyQueries` doesn't have a connector that supports proper delete (Mock connector is available but it doesn't allow us to validate if the data is deleted or updated).

Reply:
Fine. Do we have an issue for this work?
We should have TODO comment + a link here.